tests/integration/test_speakerRecognition_gradient_step_and_eval.py

```diff
@@ -0,0 +1,193 @@
+        loss_log_list.append(loss_tensor[0].item())
+
+    @pytest.mark.integration
+    def test_quartznet_speaker_reco_training(self):
+        """Integtaion test that instantiates a small QuartzNet model for speaker recognition and tests training with the
+        sample an4 data.
+        Training is run for 3 forward and backward steps and asserts that loss after 3 steps is smaller than the loss
+        at the first step.
+        """
+        with open(
+            os.path.abspath(os.path.join(os.path.dirname(__file__), "../data/quartznet_spkr_test.yaml"))
+        ) as file:
+            spkr_params = self.yaml.load(file)
+        dl = nemo_asr.AudioToSpeechLabelDataLayer(
+            # featurizer_config=self.featurizer_config,
+            manifest_filepath=self.manifest_filepath,
+            labels=None,
+            batch_size=5,
+        )
+        pre_process_params = pre_process_params = {
+            "sample_rate": 16000,
+            "normalize": "per_feature",
+            "window_size": 0.02,
+            "window_stride": 0.01,
+            "window": "hann",
+            "features": 64,
+            "n_fft": 512,
+            "frame_splicing": 1,
+            "dither": 0.00001,
+            "stft_conv": False,
+        }
+        preprocessing = nemo_asr.AudioToMelSpectrogramPreprocessor(**pre_process_params)
+        jasper_encoder = nemo_asr.JasperEncoder(**spkr_params['JasperEncoder'])
+        jasper_decoder = nemo_asr.JasperDecoderForSpkrClass(
+            feat_in=spkr_params['JasperEncoder']['jasper'][-1]['filters'],
+            num_classes=dl.num_classes,
+            pool_mode=spkr_params['JasperDecoderForSpkrClass']['pool_mode'],
+            emb_sizes=spkr_params["JasperDecoderForSpkrClass"]["emb_sizes"].split(","),
+        )
+        ce_loss = nemo_asr.CrossEntropyLossNM()
+
+        # DAG
+        audio_signal, a_sig_length, targets, targets_len = dl()
+        processed_signal, p_length = preprocessing(input_signal=audio_signal, length=a_sig_length)
+
+        encoded, encoded_len = jasper_encoder(audio_signal=processed_signal, length=p_length)
+        # logging.info(jasper_encoder)
+        log_probs, _ = jasper_decoder(encoder_output=encoded)
+        loss = ce_loss(logits=log_probs, labels=targets)
+
+        loss_list = []
+        callback = nemo.core.SimpleLossLoggerCallback(
+            tensors=[loss], print_func=partial(self.print_and_log_loss, loss_log_list=loss_list), step_freq=1
+        )
+        self.nf.random_seed = 42
+        self.nf.train(
+            [loss], callbacks=[callback], optimizer="sgd", optimization_params={"max_steps": 4, "lr": 0.002},
+        )
+        self.nf.reset_trainer()
+
+        # Assert that training loss went down
+        assert loss_list[-1] < loss_list[0]
+
+    @pytest.mark.integration
+    def test_quartznet_speaker_reco_eval(self):
+        """Integration test that tests EvaluatorCallback and NeuralModuleFactory.eval().
+        """
```

Comment:
Can you add some asserts to this test? What does this test add to your testing suite that isn't already covered?

Reply:
this is just to see if eval call back was working. Yes I agree no assertions or checks. Will remove